cloudferrylib/os/compute/nova_compute.py

```
@@ -733,31 +740,37 @@ def create_instance(self, nclient, **kwargs):
-                                                   created_instance.id,
-                                                   instance_user_id)
-        LOG.debug("Created instance '%s'", created_instance.id)
+        try:
+            curr_user_id = created_instance.user_id
+            instance_user_id = kwargs['user_id']
+            LOG.debug("Current user id = '%s' for instance '%s'",
+                      curr_user_id, created_instance.id)
+            if curr_user_id != instance_user_id:
+                if self.config.migrate.keep_usage_quotas_inst:
+                    LOG.debug("Fixing usage quota for user '%s' on '%s'",
+                              curr_user_id, instance_user_id)
+                    db = self.mysql_connector
+                    flavor_obj = nclient.flavors.get(boot_args['flavor'])
+                    flavor = {
+                        'vcpus': flavor_obj.vcpus,
+                        'ram': flavor_obj.ram,
+                    }
+                    tenant_id = created_instance.tenant_id
+                    UsageQuotaCompute\
+                        .change_usage_quota_instance(db,
+                                                     flavor,
+                                                     curr_user_id,
+                                                     instance_user_id,
+                                                     tenant_id)
+                LOG.debug("Replacing current user on '%s'", instance_user_id)
+                instances.update_user_ids_for_instance(self.mysql_connector,
+                                                       created_instance.id,
+                                                       instance_user_id)
+            LOG.debug("Created instance '%s'", created_instance.id)
+        except (sqlalchemy.exc.DisconnectionError,
+                sqlalchemy.exc.TimeoutError) as e:
+            LOG.debug(e, exc_info=True)
+            raise FailAfterCreateInstance(created_instance.id)
```

according to my idea, it is necessary to catch all exceptions to remove the instance, but not everyone agrees with this, and I limited myself to the problems with the database.